src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerStepExecution.java

```diff
@@ -28,33 +27,36 @@
     private static final transient Logger LOGGER = Logger.getLogger(ContainerStepExecution.class.getName());
     private static final transient String HOSTNAME_FILE = "/etc/hostname";
 
-    @Inject
-    private ContainerStep step;
-    @StepContextParameter private transient FilePath workspace;
-    @StepContextParameter private transient TaskListener listener;
+    private final ContainerStep step;
 
     private transient KubernetesClient client;
     private transient ContainerExecDecorator decorator;
 
+    ContainerStepExecution(ContainerStep step, StepContext context) {
+        super(context);
+        this.step = step;
+    }
+
     @Override
     public boolean start() throws Exception {
         LOGGER.log(Level.FINE, "Starting container step.");
-        StepContext context = getContext();
+        FilePath workspace = getContext().get(FilePath.class);
```

Comment:
Just curious, why do we prefer gettting `FilePath` from the context instead of having it injected?